File: serverSpring/src/main/java/com/vyatsu/auctionComplex/entity/api/Transaction.java
package com.vyatsu.auctionComplex.entity.api;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "transactions")
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    @Getter
    @Setter
    private Integer id;

    @Column(name = "amount", nullable = false)
    @Getter
    @Setter
    private BigDecimal amount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"SenderWalletId\"")
    @Getter
    @Setter
    private Wallet senderWallet;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"ReceiverWalletId\"")
    @Getter
    @Setter
    private Wallet receiverWallet;

}

File: adminWF/Classes/Transaction.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System;

namespace auctionComplex.Classes
{
    [Table("transactions")]
    internal class Transaction
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Column("amount")]
        public decimal Amount { get; set; }
        
        public Wallet SenderWallet { get; set; }

        [Column("sender_wallet_id")]
        public int SenderWalletId { get; set; }

        public Wallet ReceiverWallet { get; set; }

        [Column("receiver_wallet_id")]
        public int ReceiverWalletId { get; set; }

        [Column("timestamp")]
        public DateTime Timestamp { get; set; } 

    }
}
